src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx

```diff
@@ -46,10 +78,10 @@ export class DataSourceView extends React.Component<DataSourceViewProps, DataSou
       />
     );
 
-    let items = [];
+    let items: Array<{ name: string | undefined; disabled: boolean }> = [];
 
-    if (this.props.selectedOption) {
-      items = this.props.selectedOption.map((option) => {
+    if (this.state.selectedOption) {
+      items = this.state.selectedOption.map((option) => {
```

Comment:
Can you please explain me this - why are we using this.state for selectedOption now instead of this.props?

Reply:
we pass the selectedOption from props, there can be some empty field, e.g. `label` is optional, we will use state to reconstruct this and render with the state one

Reply:
+1, props should be enough.

Reply:
We do not only copy the value from props, but also add missing label in the state, therefore, only state.selectedOption contains sufficient information to render, props.selectedOption can miss the label for some plugins